tests/ga/test_update.py

```diff
@@ -1670,15 +1673,85 @@ def _mock_popen(cmd, *args, **kwargs):
                     with patch('azurelinuxagent.common.osutil.systemd.is_systemd', return_value=True):
                         update_handler.run(debug=True)
 
-        # Firewall-cmd should only be called 3 times - 1st to check if running, 2nd & 3rd for the QueryPassThrough cmd
-        self.assertEqual(3, len(executed_commands),
-                         "The number of times firewall-cmd should be called is only 3; Executed firewall commands: {0}; All popen calls: {1}".format(
+        # Firewall-cmd should only be called 4 times - 1st to check if running, 2nd, 3rd and 4th for the QueryPassThrough cmd
+        self.assertEqual(4, len(executed_commands),
+                         "The number of times firewall-cmd should be called is only 4; Executed firewall commands: {0}; All popen calls: {1}".format(
                              executed_commands, mock_popen.call_args_list))
         self.assertEqual(PersistFirewallRulesHandler._FIREWALLD_RUNNING_CMD, executed_commands.pop(0),
                          "First command should be to check if firewalld is running")
         self.assertTrue([FirewallCmdDirectCommands.QueryPassThrough in cmd for cmd in executed_commands],
                         "The remaining commands should only be for querying the firewall commands")
 
+    def test_it_should_set_dns_tcp_iptable_if_drop_available_accept_unavailable(self):
+        osutil._enable_firewall = True
```

Comment:
Please refrain from setting a static variable during test execution. This has the potential of breaking other unrelated test cases. The test cases should be self-contained, the mocks we use, the properties we set should ONLY be relevant inside the test, not outside. 

Please change this to use a mock instead, something like - 

`with patch.object(osutil, '_enable_firewall', True):`

Reply:
Thanks for this suggestion , I made the changes and understood the importance of not setting a static variable in test. I found other instance where _enable_firewall was set in Test in test_default, updated that as well.

Reply:
Thanks!